metrics/opencensus.go

```diff
@@ -0,0 +1,82 @@
+package metrics
+
+import (
+	pb "github.com/libp2p/go-libp2p-kad-dht/pb"
+	"go.opencensus.io/metric"
+	"go.opencensus.io/metric/metricdata"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	messageSizeBytesDistribution    = view.Distribution(0, 1, 10, 30, 70, 100, 200, 1000, 10000)
+	latencyMillisecondsDistribution = view.Distribution(0, 1, 10, 100, 300, 1000, 10000, 100000)
+)
+
+// Keys
+var (
+	KeyMessageType, _ = tag.NewKey("message_type")
+	KeyLocalPeerID, _ = tag.NewKey("local_peer_id")
+	// KeyInstanceID identifies a DHT instance by the pointer address. Useful for differentiating
+	// between different DHTs that have the same peer id.
+	KeyInstanceID, _ = tag.NewKey("instance_id")
+)
+
+func UpsertMessageType(m *pb.Message) tag.Mutator {
+	return tag.Upsert(KeyMessageType, m.Type.String())
+}
+
+const namePrefix = "libp2p_dht_kad_"
```

Comment:
Variable not required, as this won't change and if it ever does, find-replace will be perfectly acceptable.